subprojects/language-native/src/integTest/groovy/org/gradle/language/cpp/CppStaticLibraryPublishingIntegrationTest.groovy

```diff
@@ -0,0 +1,196 @@
+        def targetMachines = [machine(WINDOWS, currentArchitecture), machine(LINUX, currentArchitecture), machine(MACOS, currentArchitecture)]
+
+        given:
+        settingsFile << "include 'deck', 'card', 'shuffle'"
+        buildFile << """
+            subprojects {
+                apply plugin: 'cpp-library'
+                apply plugin: 'maven-publish'
+                
+                group = 'some.group'
+                version = '1.2'
+                publishing {
+                    repositories { maven { url '${mavenRepo.uri}' } }
+                }
+                
+                components.withType(CppComponent) {
+                    linkage = [Linkage.STATIC]
+                    targetMachines = [machines.windows().architecture('${currentArchitecture}'), machines.linux().architecture('${currentArchitecture}'), machines.macOS().architecture('${currentArchitecture}')]
+                }
+            }
+            project(':deck') { 
+                dependencies {
+                    api project(':card')
+                    implementation project(':shuffle')
+                }
+            }
+        """
+        app.deck.writeToProject(file('deck'))
+        app.card.writeToProject(file('card'))
+        app.shuffle.writeToProject(file('shuffle'))
+
+        when:
+        run('publish')
+
+        then:
+        assertMainModuleIsPublished('some.group', 'deck', '1.2', targetMachines, ["some.group:card:1.2"])
+        assertVariantsArePublished('some.group', 'deck', '1.2', ['debug', 'release'], targetMachines.findAll { it.os == currentOsFamilyName }, ["some.group:shuffle:1.2", "some.group:card:1.2"])
+
+        and:
+        assertMainModuleIsPublished('some.group', 'card', '1.2', targetMachines)
+        assertVariantsArePublished('some.group', 'card', '1.2', ['debug', 'release'], targetMachines.findAll { it.os == currentOsFamilyName })
+
+        and:
+        assertMainModuleIsPublished('some.group', 'shuffle', '1.2', targetMachines)
+        assertVariantsArePublished('some.group', 'shuffle', '1.2', ['debug', 'release'], targetMachines.findAll { it.os == currentOsFamilyName })
+
+        when:
+        def consumer = file("consumer").createDir()
+        consumer.file('settings.gradle') << ''
+        consumer.file("build.gradle") << """
+            apply plugin: 'cpp-application'
+            repositories { maven { url '${mavenRepo.uri}' } }
+            dependencies { implementation 'some.group:deck:1.2' }
+        """
+        app.main.writeToProject(consumer)
+
+        executer.inDirectory(consumer)
+        run("assemble")
+
+        then:
+        noExceptionThrown()
+        installation(consumer.file("build/install/main/debug")).exec().out == app.expectedOutput
+    }
+
+    def "can publish a library and its dependencies to a Maven repository when multiple target architectures are specified"() {
+        Assume.assumeFalse(toolChain.meets(ToolChainRequirement.WINDOWS_GCC))
```

Comment:
I think it would be good to get cygwin64 installed on the CI machines and get rid of this constraint. It seems to be appearing a fair bit.